controllers/servers/host_definer/watcher/node_watcher.py

```diff
@@ -0,0 +1,55 @@
+from queue import Empty
+from kubernetes import watch
+
+from controllers.common.csi_logger import get_stdout_logger
+from controllers.servers.host_definer.watcher.watcher_helper import NODES, Watcher
+from controllers.servers.host_definer import settings
+
+logger = get_stdout_logger()
+
+
+class NodeWatcher(Watcher):
+
+    def watch_nodes_resources(self):
+        watcher = watch.Watch()
+        for event in watcher.stream(self.core_api.list_node):
+            node_name = event[settings.OBJECT_KEY].metadata.name
+            csi_node = self._get_csi_node(node_name)
+            node_id = csi_node.node_id
+            if event[settings.TYPE_KEY] == settings.ADDED_EVENT and \
+                    self._is_csi_node_pod_deleted_while_host_definer_was_down(csi_node):
+                if self.is_host_can_be_undefined(node_name):
```

Comment:
maybe move into `_undefine_host_definitions`?

Reply:
a definition cannot be (un/)defined - it's the host that is (un/)defined.
so `_undefine_host_definitions` would be simply `_undefine_host`